### PR TITLE
Fix multiple invocations of `DefaultHttpDestinationBuilderProxyHandler#handle`

### PR DESCRIPTION
**Pull request by newtork**
## Context
 

Alternative solution to:
https://github.com/SAP/cloud-sdk-java/pull/567

Related issue:
https://github.com/SAP/cloud-sdk-java/issues/550

* Low footprint change.
* No API change.
* Limit behavior change to proxy header provider only.

## Definition of Done

- [x] Functionality scope stated & covered
- [x] Tests cover the scope above
- [x] ~Error handling created / updated & covered by the tests above~
- [x] ~Documentation updated~
- [x] Release notes updated

### CODE REVIEW

File: cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DefaultHttpDestination.java [hunk 1]
```diff
@@ -549,13 +551,29 @@ public int hashCode()
...
     public static class Builder
     {
         final List<Header> headers = Lists.newArrayList();
+
         final DefaultDestination.Builder builder = DefaultDestination.builder();
-        final DefaultHttpDestinationBuilderProxyHandler proxyHandler = new DefaultHttpDestinationBuilderProxyHandler();
+
+        @Setter( onParam = @__( { @Nullable } ), onMethod = @__( { @Nonnull } ), value = AccessLevel.PACKAGE )
+        private DefaultHttpDestinationBuilderProxyHandler proxyHandler =
+            new DefaultHttpDestinationBuilderProxyHandler();
```
Comment (newtork) [inline cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DefaultHttpDestination.java:563]: (Comment)

I wanted to keep changes for `proxyHandler` to a minimum with the _fluent, chain, setter, package-private_ accessor annotations. The change for `keyStore` and `trustStore` came "for free", drive-by.

File: cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DefaultHttpDestination.java [hunk 2]
```diff
@@ -549,13 +551,29 @@ public int hashCode()
...
     public static class Builder
     {
         final List<Header> headers = Lists.newArrayList();
+
         final DefaultDestination.Builder builder = DefaultDestination.builder();
-        final DefaultHttpDestinationBuilderProxyHandler proxyHandler = new DefaultHttpDestinationBuilderProxyHandler();
+
+        @Setter( onParam_ = @Nullable, onMethod_ = @Nonnull, value = AccessLevel.PACKAGE )
+        private DefaultHttpDestinationBuilderProxyHandler proxyHandler =
+            new DefaultHttpDestinationBuilderProxyHandler();
```
Comment (newtork) [inline cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DefaultHttpDestination.java:563]: (Comment)

I wanted to keep changes for a mutable/testable `proxyHandler` to a minimum with the _fluent, chain, setter, package-private_ accessor annotations. The change for `keyStore` and `trustStore` came "for free", drive-by.

**Review by MatKuhr — APPROVED** [review]
LGTM
